src/gtk/toga_gtk/widgets/tree.py

```diff
@@ -80,30 +52,32 @@ def change_source(self, source):
 
         self.store.clear()
 
-        def append_node(parent, root=False):
-            if self.node_has_children(parent):
-                for i, child_node in enumerate(parent):
-                    self.insert(parent, i, child_node)
-                    append_node(child_node)
+        def append_node(parent):
+            for i, child_node in enumerate(parent):
+                self.insert(parent, i, child_node)
+                append_node(child_node)
 
```

Comment:
Perhaps this is more an issue with the data sources / Node / Row, but as it is, iteration fails when their are no children since `_roots` or `_children` initialize to None. (`iter(None) => TypeError`)